### PR TITLE
GKE http endpoint

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

**Comment by piosz** [comment]
@a-robinson could you please take a look?

**Comment by piosz** [comment]
LGTM, please rebase

**Comment by a-robinson** [comment]
Sorry I missed this the last couple days. Taking a look now

File: metrics/core/labels.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ var podLabels = []LabelDescriptor{
 	LabelPodNamespace,
 	LabelPodNamespaceUID,
 	LabelLabels,
+	LabelCustomMetricName,
```
Comment (a-robinson) [inline metrics/core/labels.go]: Is this used anywhere by this PR?

File: metrics/api/v1/types/types.go [hunk 2]
```diff
@@ -44,38 +45,10 @@ type Point struct {
 // TimeseriesSchema represents all the metrics and labels.
 type TimeseriesSchema struct {
 	// All the metrics handled by heapster.
-	Metrics []MetricDescriptor `json:"metrics,omitempty"`
+	Metrics []core.MetricDescriptor `json:"metrics,omitempty"`
```
Comment (a-robinson) [inline metrics/api/v1/types/types.go]: I'm a little worried that moving the definitions of the API types into the core makes it more likely that they'll be changed in the future without considering that they're part of what needs to be a stable API.

Is there any particular motivation for this change that justifies the risk?

File: metrics/api/v1/api.go [hunk 3]
```diff
@@ -16,20 +16,38 @@ package v1
...
 
 type Api struct {
 	runningInKubernetes bool
 	metricSink          *metricsink.MetricSink
+	gkeMetrics          map[string]core.MetricDescriptor
+	gkeLabels           map[string]core.LabelDescriptor
 }
 
 // Create a new Api to serve from the specified cache.
 func NewApi(runningInKuberentes bool, metricSink *metricsink.MetricSink) *Api {
```
Comment (a-robinson) [inline metrics/api/v1/api.go]: It's an existing problem, but it wouldn't hurt to fix the "Kuberentes" typo while you're here

File: integration/heapster_api_test.go [hunk 4]
```diff
@@ -268,24 +268,28 @@ func runHeapsterMetricsTest(fm kubeFramework, svc *kube_api.Service) error {
...
-			if !exists {
-				return fmt.Errorf("timeseries: %v does not contain common label: %v", ts, label)
-			}
-		}
 		podName, podMetric := ts.Labels[core.LabelPodName.Key]
-		if podMetric {
-			for _, label := range core.PodLabels() {
+
+		/*
+			TODO(mwielgus): Uncoment once all labels are populated from POD watch
```
Comment (a-robinson) [inline integration/heapster_api_test.go]: Without having made it through the actual code changes, commenting out an existing test scares me. Does this mean the new API output won't contain everything the old API output does?

Comment (mwielgus) [inline integration/heapster_api_test.go]: We hope to have exactly same data. Just FYI - this we are rewriting Heapster so some functionality is not yet present. In the mean time we want to have the tests passing so some of them (that will obviously fail) are commented out.

File: metrics/api/v1/types/types.go (hunk 2 above)
Comment (mwielgus) [inline metrics/api/v1/types/types.go]: Ok good point.

File: metrics/core/labels.go (hunk 1 above)
Comment (mwielgus) [inline metrics/core/labels.go]: No

**Pull request merged**